client/app/bundles/course/survey/reducers/surveys.js

```diff
@@ -1,29 +1,21 @@
 import actionTypes from '../constants';
 import surveyReducer from './survey';
+import { updateOrAppend, deleteIfFound } from './utils';
 
-const initialState = [];
-
-export default function (state = initialState, action) {
-  const { type } = action;
-
-  switch (type) {
+export default function (state = [], action) {
+  switch (action.type) {
     case actionTypes.CREATE_SURVEY_SUCCESS: {
       return [...state, action.newSurveyData];
     }
     case actionTypes.UPDATE_SURVEY_SUCCESS:
     case actionTypes.LOAD_SURVEY_SUCCESS: {
-      const index = state.findIndex(survey => String(survey.id) === String(action.id));
-      return index === -1 ? [...state, action.data] :
-                            Object.assign([], state, { [index]: action.data });
+      return updateOrAppend(state, action.data);
```

Comment:
Also I think there's a need to document `action.data` somewhere, I don't understand what the state operation does without knowing the format of `action.data`. Maybe we should have a `api` folder then...